### PR TITLE
Updates Excelsior mines

### PR DESCRIPTION
**Pull request by FrownTown**
## About The Pull Request

Changed mines to look for the excelsior_rev antag_id on crossover instead of looking for the implant in the mob, because is_excelsior() is a buggy proc in this context.

Fixed mines exploding when Excelsior try to pick them up.

Fixes mines appearing on top of certain sprites.

## Why It's Good For The Game

It fixes an unintentional bug and general messy nanacode.

## Changelog
:cl:
fix: fixed Excelsior mines
/:cl:

### CODE REVIEW

**Review by genessee596 — APPROVED** [review]
nanokode good

File: code/game/objects/items/weapons/mine.dm [hunk 1]
```diff
@@ -112,8 +121,10 @@
 /obj/item/weapon/mine/Crossed(mob/AM)
 	if (armed)
 		if (isliving(AM))
+			if (AM.faction == "excelsior")
```
Comment (TorinoFermic) [inline code/game/objects/items/weapons/mine.dm:124]: There's defines with that string like ROLE_EXCELSIOR or similar, other added lines should be updated to reflect that including neutral faction

Comment (FrownTown) [reply code/game/objects/items/weapons/mine.dm]: Fair enough, I changed all the faction checks to antag_id checks and used excelsior_rev which is the string for the ROLE_EXCELSIOR define. Hopefully that works for you?

File: code/game/objects/items/weapons/mine.dm [hunk 2]
```diff
@@ -112,8 +122,11 @@
 /obj/item/weapon/mine/Crossed(mob/AM)
 	if (armed)
 		if (isliving(AM))
+			for(var/datum/antagonist/A in AM.mind.antagonist)
+				if(A.id == "excelsior_rev")
```
Comment (TorinoFermic) [inline code/game/objects/items/weapons/mine.dm:126]:
```suggestion
				if(A.id == ROLE_EXCELSIOR_REV)
```

I suggest a change from a string to a define easy to copy.

File: code/game/objects/items/weapons/mine.dm [hunk 3]
```diff
@@ -68,6 +68,16 @@
 	update_icon()
 
 /obj/item/weapon/mine/attack_hand(mob/user)
+	for(var/datum/antagonist/A in user.mind.antagonist)
+		if(A.id == "excelsior_rev" && deployed)
```
Comment (TorinoFermic) [inline code/game/objects/items/weapons/mine.dm:72]:
```suggestion
		if(A.id == ROLE_EXCELSIOR_REV && deployed)
```

**Review by TorinoFermic — APPROVED** [review]

**Pull request merged**